test/cypress/support/index.js

```diff
@@ -18,9 +18,32 @@ import './commands.js'
 import './output-logs.js'
 
 before(function () {
+  console.log('[cypress] `before`: cleaning up')
+  if (typeof navigator === 'object' && navigator.serviceWorker) {
+    cy.wrap(navigator.serviceWorker.getRegistrations()
+      .then((registrations) => {
+        console.log('[cypress] Service worker registations', registrations)
+        return Promise.all(registrations.map((registration) => {
+          // Shut down and unregister any existing service workers
+          registration.active?.postMessage({ type: 'shutdown' })
+          registration.installing?.postMessage({ type: 'shutdown' })
+          registration.waiting?.postMessage({ type: 'shutdown' })
+          return registration.unregister()
+        }))
+      }
+      )
+    )
+  }
   cy.clearCookies()
   cy.clearLocalStorage()
-  indexedDB.deleteDatabase('Group Income')
+  if (typeof indexedDB === 'object') {
+    cy.wrap(indexedDB.databases().then((db) => {
+      console.log('[cypress] Registered DBs', db)
+      return Promise.all(db.map(({ name }) => indexedDB.deleteDatabase(name)))
+    }
+    ))
+  }
+  cy.wrap(undefined).then(() => console.log('[cypress] Finished `before`'))
```

Comment:
What is this `cy.wrap(undefined)` thing and why is it here?

(Weird stuff should be either unweirded or commented)

Reply:
I can re-test or add a comment. It seemed to be needed for cypress to actually wait for the console.log instead of printing it before being actually done (i.e., forces the call into a queue).

Reply:
Yeah that makes sense, still a comment is needed. And maybe check if `cy.then()` works directly instead of needing to wrap something (or if there's a similar/related API that doesn't require `.wrap()`)

Reply:
Documented it.